syntax = "proto3";

package studio.assets;
import "google/protobuf/timestamp.proto";

message AssetMetadata {
  string id = 1;
  string name = 2;
  string type = 3;
  int64 version = 4;
  string description = 5;
  google.protobuf.Timestamp created_at = 6;
}

message ListAssetsRequest {}

message ListAssetsResponse {
  repeated AssetMetadata assets = 1;
}

message GetAssetRequest {
  string id = 1;
}

message GetAssetResponse {
  bytes data = 1;
  string format = 2;
}

service AssetService {
  rpc ListAssets(ListAssetsRequest) returns (ListAssetsResponse);
  rpc GetAsset(GetAssetRequest) returns (GetAssetResponse);
}
